packages/core/src/utils/should-ignore-span.ts

```diff
@@ -0,0 +1,67 @@
+import type { ClientOptions } from '../types-hoist/options';
+import type { SpanJSON } from '../types-hoist/span';
+import { isMatchingPattern, stringMatchesSomePattern } from './string';
+
+/**
+ * Check if a span should be ignored based on the ignoreSpans configuration.
+ */
+export function shouldIgnoreSpan(
+  span: Pick<SpanJSON, 'description' | 'op'>,
+  ignoreSpans: ClientOptions['ignoreSpans'],
+): boolean {
+  if (!ignoreSpans?.length) {
+    return false;
+  }
+
+  if (!span.description) {
+    return false;
+  }
+
+  // First we check the simple string/regex patterns - if the name matches any of them, we ignore the span
+  const simplePatterns = ignoreSpans.filter(isStringOrRegExp);
+  if (simplePatterns.length && stringMatchesSomePattern(span.description, simplePatterns)) {
+    return true;
+  }
+
+  // Then we check the more complex patterns, where both parts must match
+  for (const pattern of ignoreSpans) {
+    // Have already checked for simple patterns, so we can skip these
+    if (isStringOrRegExp(pattern) || (!pattern.name && !pattern.op)) {
+      continue;
+    }
```

Comment:
l: I think performance-wise, it makes more sense to handle everything in one for loop here, no? If users didn't pass simple patterns or none of them matched, we'd iterate twice over the ignoreSpans array. Also avoids calling `isStringOrRegExp` twice in this case.
(not the end of the world but it scales with the number of spans...)